R/rstudioapi_util.R

```diff
@@ -226,7 +242,22 @@ update_addin_registry <- function(addin_registry) {
     jsonlite::write_json(addin_descriptions_flat, addin_registry, pretty = TRUE)
 }
 
+update_addin_registry_safely <- function(addin_registry) {
+    if (!rstudioapi_enabled()) return(NULL)
+
+    tryCatch(update_addin_registry(addin_registry),
```

Comment:
I suggest that the tryCatch be used per `read.dcf` so that a single malformed file won't make the entire feature broken.

Reply:
Just notice that you check the number fields at https://github.com/Ikuyadeu/vscode-R/pull/422/files#diff-11107abea790174063b25c459ad87e095af4cab1fbf98bfe33338337901a5939R221-R223. If the malformed file won't cause `read.dcf` to stop, then a single malformed won't make the feature broken?

Reply:
Yeah 
  * if it's malformed short in columns - return null and it will be filtered out by rbind
  * if it's malformed long - take the first 4 (and append 1)
  * if there's something that causes an error like it's NULL and I can't index it, error

Reply:
I like your idea of moving the tryCatch per file read/parse

Reply:
I am just worried they're could be some error case on the bind that I haven't thought of

Reply:
It seems that even if all are NULL rbind and write_json will work. So let's do it your way, that's better because I can report the specific file.